app/src/main/java/com/example/spacex_demo_app/MainActivity.kt

```diff
@@ -19,17 +26,31 @@ class MainActivity : AppCompatActivity() {
 
         val apolloClient =
             ApolloClient.builder().serverUrl(BuildConfig.SPACEX_ENDPOINT).build()
-        var text = ""
         disposable = CompositeDisposable()
-        disposable?.add(SpaceXApi(apolloClient).getLaunches().subscribe { resp ->
-            text += resp.data?.launches?.get(0)?.details?.take(lim)
-            textView.text = text
-        })
-
-        disposable?.add(SpaceXApi(apolloClient).getLaunchById("9").subscribe { resp ->
-            text += resp.data?.launch?.details?.take(lim)
-            textView.text = text
-        })
+
+        val button: Button = findViewById(R.id.button)
+        val progressBar: ProgressBar = findViewById(R.id.progressBar)
+        button.setOnClickListener {
+            it.visibility = View.GONE
+            progressBar.visibility = View.VISIBLE
```

Comment:
Move it to Rx chain. Use operator that triggers before beginning to emit data
`it.visibility = View.GONE
 progressBar.visibility = View.VISIBLE`